pkg/collector/corechecks/cluster/kubernetes_state.go

```diff
@@ -421,3 +480,26 @@ func formatMetricName(name string) string {
 	log.Tracef("KSM metric '%s' is not found in the metric names mapper", name)
 	return ksmMetricPrefix + name
 }
+
+// resourceNameFromMetric returns the resource name based on the metric name
+// It relies on the conventional KSM naming format kube_<resource>_suffix
+// returns an empty string otherwise
+func resourceNameFromMetric(name string) string {
```

Comment:
Wanted to double check that this does the right thing for the `resourcequota` metrics which have this weird naming e.g. `kubernetes_state.resourcequota.count/ingresses.extensions.limit` (this is the outgoing metric name, I am not quite sure what the source for it is. Mind double checking?

Reply:
this function handles the original ksm metrics `kube_<resource>_suffix` (for resourcequota see here https://github.com/kubernetes/kube-state-metrics/blob/master/docs/resourcequota-metrics.md).

`kubernetes_state.resourcequota.count/ingresses.extensions.limit` is a Datadog metric generated by a transformer, and has nothing to do with this function. fwiw the `/` is replaced by a `_` when the metric is processed in Datadog

Reply:
Ah yes, totally, sorry I was confused.

Reply:
np, thank you!